### PR TITLE
Allow to specify IAM instance profiles on workers

### LINKED ISSUE #143: Allow to specify IAM instance profiles on workers
**How to categorize this issue?**

/area security
/kind api-change
/priority normal
/platform aws

**What would you like to be added**:

The ability to specify IAM instance profile on the workers. e.g. :

```yaml
apiVersion: extensions.gardener.cloud/v1alpha1
kind: Worker
metadata:
  name: worker
  namespace: shoot--foobar--aws
spec:
  type: aws
  region: eu-west-1
  secretRef:
    name: cloudprovider
    namespace: shoot--foobar--aws
  pools:
  - name: cpu-worker
    machineType: m4.large
    machineImage:
      name: coreos
      version: 2135.6.0
    zones:
    - eu-west-1a
  providerConfig:
    apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
    kind: WorkerConfig
    volume:
      iops: 10000
    dataVolumes:
    - name: kubelet-dir
      snapshotID: snap-13234
    iamInstaceProfile:
      name: my-profile
      # arn: my-instance-profile-arn # specify either ARN or name.
```

**Why is this needed**:

This would allow to eventually (optionally) remove the extension's dependency to the IAM API, reducing the attack surface.

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area usability
/kind enhancement
/platform aws
/merge squash

**What this PR does / why we need it**:
Enable end-users to overwrite the IAM instance profile used for the machines in case the default IAM policy does not fit their use-cases.

**Which issue(s) this PR fixes**:
Fixes #143

**Special notes for your reviewer**:
- ✅ ~Depends on https://github.com/gardener/machine-controller-manager-provider-aws/pull/34 and needs to wait until it is merged and released.~
- The `eu.gcr.io/gardener-project/gardener/extensions/provider-aws:v1.25.0-dev-rf2` image can be used for local testing.

**Release note**:

``` feature operator github.com/gardener/machine-controller-manager-provider-aws #32 @rfranzke
It's now possible to properly use the `IAM.ARN` field in the `MachineClass` specification. Earlier, it was only possible to use `IAM.Name`.
```
```feature user
It's now possible to specify which IAM instance profile should be used for the machines of a particular worker pool. The respective name or ARN can be set via the `iamInstanceProfile.{name,arn}` fields in the `WorkerConfig`.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
FYI/off-topic: I had to manually update the `machine-controller-manager-provider-aws` image as the automatic update PRs seem to be currently broken (https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-extension-provider-aws-updates-master/jobs/master-create_update_pull_requests-job/builds/14640), I suspect due to #327 (cc @schrodit @AndreasBurger).

File: pkg/apis/aws/validation/worker_test.go [hunk 1]
```diff
@@ -189,5 +193,78 @@ var _ = Describe("ValidateWorkerConfig", func() {
...
+
+				errorList := ValidateWorkerConfig(worker, rootVolumeIO1, dataVolumes, fldPath)
+
+				Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal("config.iamInstanceProfile.arn"),
+				}))))
+			})
+
+			It("should forbid specifying a valid name", func() {
```
Comment (noqu) [inline pkg/apis/aws/validation/worker_test.go:249]: You probably mean `allow` here? Same below?

**Review by noqu — COMMENTED** [review]
Thanks a lot, this looks as if it was exactly what we need.

**Review by prashanth26 — DISMISSED** [review]
Minor typos.
/lgtm otherwise

**Review by prashanth26 — APPROVED** [review]
/lgtm
/squash

**Pull request merged**